### PR TITLE
Implement serializers strategies to save the message on repository

### LINKED ISSUE #39: Change the strategy of saving messages on repo (remove proto-buf)
The actual solution uses protobuf-net but it creates a dependency to the client.
IT should be changed to the old solution using serialize, encoder and zipper.

### PR DESCRIPTION
**Pull request by martinhonovais**
# Description

We have implemented a new configuration that allows the developer to choose the way that the messages should be saved on the repository when the developer is using the retry durable policy.

There were implemented two strategies:
1. Using ProtobufNet to serialize the messages (Removed)
2. Using NewtonsoftJson to serialize the messages,

Fixes #39 

## How Has This Been Tested?

We have changed the one integration test to use NewtonsoftJson strategy and another one to use ProtobufNet strategy

## Checklist

-   [X] My code follows the style guidelines of this project
-   [X] I have performed a self-review of my own code
-   [X] I have added tests to cover my changes
-   [ ] I have made corresponding changes to the documentation

### Disclaimer

By sending us your contributions, you are agreeing that your contribution is made subject to the terms of our [Contributor Ownership Statement](https://github.com/Farfetch/.github/blob/master/COS.md)

### CODE REVIEW

File: src/KafkaFlow.Retry/Durable/Definitions/Builders/RetryDurableEmbeddedClusterDefinitionBuilder.cs [hunk 1]
```diff
@@ -109,15 +110,15 @@ RetryDurablePollingDefinition retryDurablePollingDefinition
...
+                        .WithAcks(Acks.Leader)
                 )
                 .AddConsumer(
                     consumer => consumer
                         .Topic(this.retryTopicName)
                         .WithGroupId(consumerGroupId)
                         .WithBufferSize(this.retryConsumerBufferSize)
                         .WithWorkersCount(this.retryConsumerWorkersCount)
-                        .WithAutoOffsetReset(AutoOffsetReset.Latest) // TODO: this settings should be reviewed
+                        .WithAutoOffsetReset(AutoOffsetReset.Earliest)
```
Comment (rodrigobelo) [inline src/KafkaFlow.Retry/Durable/Definitions/Builders/RetryDurableEmbeddedClusterDefinitionBuilder.cs:121]: Do we have idempotency protection?

File: src/KafkaFlow.Retry/Durable/MessageSerializerStrategy.cs [hunk 2]
```diff
@@ -0,0 +1,7 @@
+﻿namespace KafkaFlow.Retry
+{
+    public enum MessageSerializerStrategy
+    {
+        NewtonsoftJson = 1
```
Comment (rodrigobelo) [inline src/KafkaFlow.Retry/Durable/MessageSerializerStrategy.cs:5]: As a good practice, always have enum default values such as "None" or "Unknown".

**Review by rodrigobelo — COMMENTED** [review]
I'm a bit worried about public accessible methods that are not checking parameters consistency such as nullability allowances. I've spotted some in this review.

File: src/KafkaFlow.Retry/Durable/MessageSerializerStrategy.cs (hunk 2 above)
Comment (martinhonovais) [reply src/KafkaFlow.Retry/Durable/MessageSerializerStrategy.cs:5]: 👍 Done.

File: src/KafkaFlow.Retry/Durable/Definitions/Builders/RetryDurableEmbeddedClusterDefinitionBuilder.cs (hunk 1 above)
Comment (martinhonovais) [reply src/KafkaFlow.Retry/Durable/Definitions/Builders/RetryDurableEmbeddedClusterDefinitionBuilder.cs:121]: As we talked, I'll open a issue to think about it in the next moves.

**Review by rodrigobelo — APPROVED** [review]

**Pull request merged**